stumpy/core.py

```diff
@@ -2254,14 +2375,29 @@ def rolling_isconstant(a, w):
     w : numpy.ndarray
         The rolling window size
 
+    custom_func : object, default None
+        A custom, user-defined function that returns boolean numpy ndarray that indicate
+        if a subsequence is constant or not. It takes two arguments, `a`, a 1-D array,
+        and `w`, the window size, and may have default arguments if needed. When `None`,
+        this will be default to the function `core._rolling_isconstant`.
+
     Returns
     -------
     output : numpy.ndarray
         Rolling window isconstant.
     """
+    rolling_isconstant_func = _rolling_isconstant
+    if custom_func is not None:
+        custom_func_args = set(inspect.signature(custom_func).parameters.keys())
+        if len(set(["a", "w"]).difference(custom_func_args)):  # pragma: no cover
+            msg = "Incompatible parameters found in `custom_func`"
+            warnings.warn(msg)
+        else:
+            rolling_isconstant_func = custom_func
+
     axis = a.ndim - 1
     return np.apply_along_axis(
-        lambda a_row, w: _rolling_isconstant(a_row, w), axis=axis, arr=a, w=w
+        lambda a_row, w: rolling_isconstant_func(a_row, w), axis=axis, arr=a, w=w
```

Comment:
I understand. However, I would prefer to keep `rolling_isconstant` pure because the assumption is that the responsibility of this function is to look for constant subsequences. 

Outside of `core.rolling_isconstant`, I would simply add another function called `core.check_isconstant`, which simply does np.logical_and(`T_subseq_isconstant`, `T_subseq_isfinite`) raises a ValueError (describing the precise index where things are bad). In this case, I would rather not save the user and force/train them to fix the problem. I think this is fine since the error is caught very early in the execution process. It would really suck to raise an error after 10 minutes of computing the matrix profile.

So, `core.rolling_isconstant` will blindly return an array if an array is provided or apply a custom function if one is provided (i.e., blindly trust the user). Otherwise, it will apply `core._rolling_isconstant`.
Then, we check it `T_subseq_isconstant` separately.

Reply:
On second thought, I would accept a function called `core.fix_isconstant_isfinite_conflicts` that fixes the conflicts between `T_subseq_isconstant` and `T_subseq_isfinite` and maybe it has the following signature:

```
def fix_isconstant_isfinite_conflicts(T, m, T_subseq_isconstant, T_subseq_isfinite=None):
    if T_subseq_isfinite is None:
        T_subseq_isfinite = rolling_isfinite(T, m)

    fixed = np.logical_and(T_subseq_isconstant, T_subseq_isfinite)
    # Add warning where `fixed != T_subseq_isconstant` and provide indices
    return fixed
```

Something like that would be fine too

Reply:
Thanks for the suggestion. I think I got the point.